mods/sethome/init.lua

```diff
@@ -2,64 +2,59 @@ local homes_file = minetest.get_worldpath() .. "/homes"
+	if input then
 		repeat
-            local x = input:read("*n")
-            if x == nil then
-            	break
-            end
-            local y = input:read("*n")
-            local z = input:read("*n")
-            local name = input:read("*l")
-            homepos[name:sub(2)] = {x = x, y = y, z = z}
-        until input:read(0) == nil
-        io.close(input)
-    else
-        homepos = {}
-    end
+			local x = input:read("*n")
+			if not x then
+				break
+			end
+			local y = input:read("*n")
+			local z = input:read("*n")
+			local name = input:read("*l")
+			homepos[name:sub(2)] = {x = x, y = y, z = z}
+		until not input:read(0)
+		io.close(input)
+	end
 end
 
 loadhomes()
 
 minetest.register_privilege("home", "Can use /sethome and /home")
 
-local changed = false
-
 minetest.register_chatcommand("home", {
-    description = "Teleport you to your home point",
-    privs = {home=true},
-    func = function(name)
-        local player = minetest.get_player_by_name(name)
-        if player == nil then
-            -- just a check to prevent the server crashing
-            return false
-        end
-        if homepos[player:get_player_name()] then
-            player:setpos(homepos[player:get_player_name()])
-            minetest.chat_send_player(name, "Teleported to home!")
-        else
-            minetest.chat_send_player(name, "Set a home using /sethome")
-        end
-    end,
+	description = "Teleport you to your home point",
+	privs = {home = true},
+	func = function(name)
+		local player = minetest.get_player_by_name(name)
+		if not player then
+			-- just a check to prevent the server crashing
+			return false
+		end
+		if homepos[name] then
+			player:setpos(homepos[name])
+			minetest.chat_send_player(name, "Teleported to home!")
+		else
+			minetest.chat_send_player(name, "Set a home using /sethome")
+		end
+	end,
 })
 
 minetest.register_chatcommand("sethome", {
-    description = "Set your home point",
-    privs = {home=true},
-    func = function(name)
-        local player = minetest.get_player_by_name(name)
-        local pos = player:getpos()
-        homepos[player:get_player_name()] = pos
-        minetest.chat_send_player(name, "Home set!")
-        changed = true
-        if changed then
-        	local output = io.open(homes_file, "w")
-            for i, v in pairs(homepos) do
-                output:write(v.x.." "..v.y.." "..v.z.." "..i.."\n")
-            end
-            io.close(output)
-            changed = false
-        end
-    end,
+	description = "Set your home point",
+	privs = {home = true},
+	func = function(name)
+		local player = minetest.get_player_by_name(name)
+		if not player then
+			-- another check to prevent the server crashing
+			return false
+		end
+		homepos[name] = vector.round(player:getpos())
+		minetest.chat_send_player(name, "Home set!")
+		local output = io.open(homes_file, "w")
+		for i, v in pairs(homepos) do
+			output:write(v.x.." "..v.y.." "..v.z.." "..i.."\n")
```

Comment:
You have 7 concatenations per write here and then loop that through all saved positions, passing it to `:write()` each time.
If it weren't in a loop a `string.format` would be most appropriate, but since this can add up to hundreds of entries for a server (which would mean thousands of concatenations you're doing here) i'd rather go for a `table.concat` here ending in a single `:write()`-call.